### DISCUSSION TITLE
Distribution of material properties

### DISCUSSION THREAD

**Discussion by rnpratoori** [idaholab/moose#29592] [Q&A Modules: General]
Hello team,

I want to create a material property where I know the domain average and I want to create a distribution of the value with a maximum and a minimum value (similar to how its done in `RandomIC`). Also, I want to use this value to calculate another material property, maybe using `DerivativeParsedMaterial`. What is the best way to approach this? I did look at Example 08, but I am not sure if it will fit my use case.

Best,
Raghunandan.

**Comment by GiudGiud** [comment]
hello

This is a "non local" material because the initialization depends on values which are not local. This wont be possible on-the-fly, you will need an initial step where you sample the value of the material (using a [distribution](https://mooseframework.inl.gov/syntax/Distributions/index.html)) then compute its integral, then renormalize

I would compute the integral of the material in a ElementUserObject and set the local values of the material there, then the Material only uses a pointer to the user object to retrieve the value at each quadrature point.

**Reply by rnpratoori** [reply]
Hello @GiudGiud. Thank you for your response.

I am not sure I quite understand what you mentioned. Is there an example for something similar or some documentation? I could not find anything in the documentation regarding ElementUserObject.

**Reply by GiudGiud** [reply]
I dont think there's an example.
@lynnmunday you just made some nonlocal coupling material iirc did you have a good example?

**Comment by lynnmunday** [comment]
@GiudGiud I don't remember doing this. 
@rnpratoori Something you could try is create random initial conditions for an auxvariable using RandomIC and then rescale and turn this auxVariable into a material using CoupledValueFunctionMaterial.  I'm assuming you already know your maximum and minimum values.  There are also some other RandomIC's you can look at if you don't want a uniform sampling form 0 to 1 like the VolumeWeightedWeibull

**Reply by GiudGiud** [reply]
I was thinking about this work #29576 for considering nonlocal dependencies in a material but I did not dive into it either so maybe it's different

**Reply by lynnmunday** [reply]
I see.  If @rnpratoori needs to compute some kind of correlated nonlocal distribution then they could use some of the averaging from that PR you mentioned to create the element userObject like you recommended.  That PR is using a vectorpostprocessor but that should be similar to what is needed for a userObject.  RandomMaterial might also do what they want.
